networking/networkConfigMonitor.go

```diff
@@ -140,6 +140,10 @@ func (inst *networkManagerInstance) ProcessDhcpConfigChange(fieldchanged string,
 
 //Function to process if config change
 func (inst *networkManagerInstance) ProcessIfConfigChange(fieldchanged string, futvalue interface{}, curvalue interface{}, index int) {
+	//if we need a new target
+	if index >= len(inst.networkConfig.Interfaces) {
```

Comment:
I now believe this might be covering up a bug elsewhere.  Shouldn't this array be non-empty after reading the config.yaml here: https://github.com/armPelionEdge/maestro/blob/458a91f7278e8431ab446c3b751500e2b390c262/networking/manager.go#L2302